ksqldb-engine/src/main/java/io/confluent/ksql/util/PersistentQueryMetadata.java

```diff
@@ -215,5 +220,13 @@ public synchronized void restart() {
    */
   public synchronized void stop() {
     doClose(false);
+    scalablePushRegistry.ifPresent(ScalablePushRegistry::close);
+  }
+
+  /**
+   * Gets the scalable push query registry, if one was created for this persistent query.
+   */
+  public Optional<ScalablePushRegistry> getScalablePushRegistry() {
```

Comment:
This is more or less for the other PR together: since I did not see the caller to construct this object I guess it is in other PRs, my question is though how do we find which `PersistentQueryMetadata` to pass in when we build the builder? That is because, it seems like the trace are:

`PushPhysicalPlanBuilder#persistentQueryMetadata -> PeekStreamOperator#scalablePushRegistry -> scalablePushRegistry#register`

And we would have one registry for each source stream/table only, where all registered push queries would be given the record from the `peek` call. This means we must be able to get the corresponding `persistentQueryMetadata` to pass to the `PushPhysicalPlanBuilder`, but at that time how we would know which source stream/table it is for?

Reply:
It's actually in this PR.  There's a factory method called `ScalablePushRegistry.create`.

If you look in `QueryExecutor.applyScalablePushProcessor`, you can see that when the `PersistentQueryMetadata` is being constructed, that's when this is created, so we know exactly how the two match up.

Reply:
Okay, I guess the confusion from my side is that I did not see where the constructor of `PushPhysicalPlanBuilder` is called in either PR, and hence hard to guess which `persistentQueryMetadata` parameter would be passed in. If you have that constructor called somewhere could you point me to it?